app/models/concerns/aliquotable.rb

```diff
@@ -50,19 +50,49 @@ def available_volume
       primary_aliquot.volume - used_volume
     end
 
-    # Method to check if there is enough volume available.
-    # It compares the available volume with the required volume.
+    # Method: is_available_volume_sufficient
     #
-    # @param required_volume [Numeric] The volume required for the operation.
+    # This method is used to validate if the available volume is sufficient for a required volume.
     #
-    # @return [Boolean] Returns true if the available volume is greater than
-    # or equal to the required volume.
-    # If not, it adds an error message to `errors` and returns false.
-    def volume_check(required_volume)
+    # The method performs the following checks:
+    # 1. If the available volume is greater than or equal to the required volume,
+    # the method returns true.
+    # 2. If the available volume is less than the required volume, the method adds an error to the
+    # base attribute of the record and returns false.
+    #
+    # @param required_volume [Numeric] The volume that is required.
+    #
+    # @return [Boolean] Returns true if the available volume is sufficient, false otherwise.
+    def available_volume_sufficient(required_volume)
       return true if available_volume >= required_volume
 
       errors.add(:base, 'Insufficient volume available')
       false
     end
+
+    # Method: primary_aliquot_volume_sufficient
+    #
+    # This method is used to validate the volume of the primary aliquot in the library.
+    # It is typically used as a callback before updating a library record.
+    #
+    # The method performs the following checks:
+    # 1. If the primary aliquot has not changed its volume, the method returns immediately
+    # without performing any further checks.
+    # 2. If the volume of the primary aliquot is greater than or equal to the used volume,
+    # the method returns true.
+    # 3. If the volume of the primary aliquot is less than the used volume,
+    # the method adds an error to the library record and aborts the update operation.
+    #
+    # @return [nil, true, false] Returns nil if the primary aliquot has not changed its volume,
+    #  true if the volume of the primary aliquot is greater than or equal to the used volume, and
+    # false if the volume of the primary aliquot is less than the used volume.
+
+    def primary_aliquot_volume_sufficient
+      return unless primary_aliquot&.volume_changed?
```

Comment:
Not sure this check (line 91) should be here. If we add a used_aliquot but don't change the primary_aliquot we still want to check the primary aliquot has sufficient volume.

Reply:
I thought it will be covered by available_volume_sufficient method.

Please correct, if I am wrong. My understanding is

The `primary_aliquot_volume_sufficient` is called when library volume is updated/edited (attached to library model hook before_update :primary_aliquot_volume_sufficient) 
and  available_volume_sufficient is called when a used_aliquot value is updated (e.g when a library added  to pool)

Reply:
Ah yes good point. I think when a library is added to a pool we could call valid on the library as now it has the used_aliquot from the pool. Ill look into it in DPL-1072 and update accordingly.

Reply:
Yes, with the new changes, the concept of "primary_aliquot_volume_sufficient" is somewhat obsolete. 

Currently, the UI prevents the user from editing a value less than the primary aliquot volume, but the user experience aspect of this restriction needs to be explored further. It may limit user interaction, especially when typing into the volume text field. 
This is something you can also check as part of the review process, please.
 If this restriction proves to be problematic for the user experience, we may need to reconsider its implementation and potentially retain both functionalities `used_volume` and `primary_aliquot_volume_sufficient` check.